src/transformStateWithClones.js

```diff
@@ -8,6 +8,48 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const stateCopy = { ...state };
+  const stateHistory = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        stateHistory.push(addProperties(stateCopy, action.extraData));
+        break;
+      case 'removeProperties':
+        stateHistory.push(removeProperties(stateCopy, action.keysToRemove));
+        break;
+      case 'clear':
+        stateHistory.push(clearState(stateCopy));
```

Comment:
You should not push the state copy to the stateHistory array within each case. Instead, modify the stateCopy according to the action and then push the stateCopy to the stateHistory array after the switch statement, within the loop but outside of the switch block. This adheres to the DRY principle and ensures that the push operation is not repeated unnecessarily.